crates/autopilot/src/arguments.rs

```diff
@@ -220,6 +220,51 @@ pub struct Arguments {
     /// `order_events` database table.
     #[clap(long, env, default_value = "30d", value_parser = humantime::parse_duration)]
     pub order_events_cleanup_threshold: Duration,
+
+    /// List of addresses which receives a specific discount for the protocol
+    /// fees
+    #[clap(long, env, use_value_delimiter = true)]
+    pub protocol_fee_discounted_addresses: Vec<ProtocolFeeDiscountedAddresses>,
+}
+
+#[derive(Debug, Clone)]
+pub struct ProtocolFeeDiscountedAddresses {
+    pub address: H160,
+    pub kind: ProtocolFeeDiscountKind,
+}
+
+impl FromStr for ProtocolFeeDiscountedAddresses {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
I'm tempted to say let's not over-engineer features and only add different discounts when we actually need them.